### PR TITLE
Adds proto for the GatewayClass based on the GAMMA Kubernetes Sig

### PR DESCRIPTION
**Pull request by missylbytes**
### Description

Adds GatewayClass proto based on the [GAMMA Kubernetes Sig](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1alpha2.GatewayClass)

### Testing & Reproduction steps
👀

### CODE REVIEW

File: proto-public/pbmesh/v2beta1/gateway_class.proto [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+//
+// This is a Resource type.
+message GatewayClass {
+  option (hashicorp.consul.resource.spec) = {scope: SCOPE_CLUSTER};
+
+  // ControllerName is the name of the controller that manages Gateways of this class
+  string controller_name = 1;
+
+  // ParametersRef is a reference to a resource that contains the configuration
+  // parameters corresponding to the GatewayClass.
```
Comment (t-eckert) [inline proto-public/pbmesh/v2beta1/gateway_class.proto:23]:
```suggestion
  // ParametersRef refers to a resource responsible for configuring 
  // the behavior of the GatewayClass.
```

File: proto-public/pbmesh/v2beta1/gateway_class.proto [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  // Description of GatewayClass
+  string description = 3;
+}
+
+// NOTE: this should align to the GAMMA/gateway-api version, or at least be
+// easily translatable.
+//
+// ParametersReference specifies a resource that contains controller-specific configuration
+// for a resource
```
Comment (t-eckert) [inline proto-public/pbmesh/v2beta1/gateway_class.proto:34]: Again, Kubernetes controller, just to steer as far away from confusion as humanly possible.

**Review by t-eckert — APPROVED** [review]
You're a proto pro!

File: proto-public/pbmesh/v2beta1/gateway_class.proto [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v2beta1;
+
+import "pbresource/annotations.proto";
+
+// NOTE: this should align to the GAMMA/gateway-api version, or at least be
+// easily translatable.
+//
+// https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1alpha2.GatewayClass
```
Comment (nathancoleman) [inline proto-public/pbmesh/v2beta1/gateway_class.proto]: Can reference `v1` resource here

```suggestion
// https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.GatewayClass
```

**Review by nathancoleman — APPROVED** [review]

File: proto-public/pbmesh/v2beta1/gateway_class.proto (hunk 1 above)
Comment (missylbytes) [reply proto-public/pbmesh/v2beta1/gateway_class.proto]: Done!

File: proto-public/pbmesh/v2beta1/gateway_class.proto (hunk 2 above)
Comment (missylbytes) [reply proto-public/pbmesh/v2beta1/gateway_class.proto]: Done

**Pull request merged**